integrated_channels/sap_success_factors/client.py

```diff
@@ -31,28 +31,91 @@ def get_oauth_access_token(url_base, client_id, client_secret, company_id, user_
-                    'userType': 'user',
+                    'userType': 'admin',
                     'resourceType': 'learning_public_api',
                 }
             },
-            auth=(client_id, client_secret)
+            auth=(client_id, client_secret),
+            headers={'content-type': 'application/json'}
         )
 
+        response.raise_for_status()
         data = response.json()
-
         try:
-            return data['access_token'], datetime.datetime.utcfromtimestamp(data['expiresIn'])
-        except KeyError:
+            return data['access_token'], datetime.datetime.utcfromtimestamp(data['expires_in'] + int(time.time()))
+        except Exception:
             raise requests.RequestException(response=response)
+
+    def __init__(self, url_base, oauth_access_token, timeout=5):
+        """
+        Instantiate a new client.
+
+        Raises:
+            ValueError: If a URL or access token are not provided.
+        """
+
+        if not url_base:
+            raise ValueError('An API url must be supplied!')
+
+        if not oauth_access_token:
+            raise ValueError('An API access token must be supplied!')
+
+        self.url_base = url_base
+
+        session = requests.Session()
+        session.timeout = timeout
+        session.headers['Authorization'] = 'Bearer {}'.format(oauth_access_token)
+        self.session = session
+        super(SAPSuccessFactorsAPIClient, self).__init__()
+
+    def send_completion_status(self, payload):
+        """
+        Send a completion status payload to the SuccessFactors OCN Completion Status endpoint
+
+        Args:
+            payload (str): JSON object (serialized from LearnerDataTransmissionAudit)
+                containing completion status fields per SuccessFactors documentation.
+
+        Returns:
+            The body of the response from SAP SuccessFactors, if successful
+        Raises:
+            HTTPError: if we received a failure response code from SAP SuccessFactors
+        """
+        global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
+        url = self.url_base + global_sap_config.completion_status_api_path
+        response = self.session.post(url, json=payload)
+        response.raise_for_status()
+        return response.json()
+
+    def send_course_import(self, payload):
+        """
+        Send courses payload to the SuccessFactors OCN Course Import endpoint
+
+        Args:
+            payload (str): JSON object containing course import data per SuccessFactors documentation.
+
+        Returns:
+            The body of the response from SAP SuccessFactors, if successful
+        Raises:
+            HTTPError: if we received a failure response code from SAP SuccessFactors
+        """
+        global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
+        url = self.url_base + global_sap_config.course_api_path
+        response = self.session.post(url, json=payload)
```

Comment:
yes - i was moving towards this direction :-)
i'll be updating this PR momentarily